### ISSUE TITLE
401 Problem

### ISSUE THREAD

**Issue by semihyilmaz** [UniversityPCCare/gsx-rest-php#4]
I am getting Error 401. I also created a new certificate from Apple. How do we solve this problem?
Tried to fetch Auth Token for user xxxx.
My problem is some : https://forums.developer.apple.com/thread/126747

**Comment by kevingrabher** [comment]
Have you tried via curl?

`curl -X GET https://partner-connect-uat.apple.com/gsx/api/authenticate/check -H 'X-Apple-ShipTo: 0000123456' -H 'X-Apple-SoldTo: 0000123456' --cert AppleCare-Partner-0000123456.Test.apple.com.chain.pem:Password123 --key privatekey-test.pem`

Are you certain your IP address is whitelisted?
Be sure to use the correct certificate and key for the test environment.

If nothing helps you'll perhaps need to contact GSX WS support.

**Comment by upccjesse** [comment]
That was actually my post on the Apple forums, @semihyilmaz.

I ended up doing two things, but I think only one thing properly fixed it:

1. Had Apple regenerate my certificate.
2. Realized that there was some incompatibility with OpenSSL+NSS and I had to run a command to convert my certificate. I don't recall what that command was off the top of my head and I'm not in a position to look for my notes at the moment but a quick Google search turns up this command which _I think_ is what I used: `openssl rsa -des3 -in your.key -out your.encrypted.key` (See https://stackoverflow.com/questions/22499425/ssl-certificate-generated-with-openssl-not-working-on-nss)

Aside from that, double check what @kevingrabher suggested and otherwise contact GSX WS team.
